### PR TITLE
[RPD-234] [BUG] Handle existing matcha configuration better

### PR DESCRIPTION
**Pull request by Christopher-Norman**
When a user deletes a resource group manually or uses a shared state and another person has run `matcha destroy full` the user would be left with an out-of-sync config file. This would cause errors when running `matcha provision` again as the specified resource group would not exist.

This PR aims to fix this by checking if the state is stale (the user has a remote state config file but the resource group does not exist) and based on the user's decision, either:

1. Removes the remote config file and continue with a new deployment
2. Or the user can stop the command at this point

## Checklist

Please ensure you have done the following:

* [x] I have read the [CONTRIBUTING](https://github.com/fuzzylabs/matcha/blob/main/CONTRIBUTING.md) guide.
* [x] I have updated the documentation if required.
* [x] I have added tests which cover my changes.

## Type of change

Tick all those that apply:

* [x] Bug Fix (non-breaking change, fixing an issue)
* [ ] New feature (non-breaking change to add functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Other (add details above)

### CODE REVIEW

File: src/matcha_ml/cli/provision.py [hunk 1]
```diff
@@ -75,6 +75,20 @@ def provision_resources(
...
 
+    if remote_state_manager.is_state_stale():
+        print(
+            "Matcha has detected a stale state file. This means that your local configuration is out of sync with the remote state, the resource group may have been removed."
+        )
+        remove_local_config = typer.confirm(
+            "Do you want to remove the existing local config and continue?"
+        )
+        if not remove_local_config:
+            raise typer.Exit()
```
Comment (dudeperf3ct) [inline src/matcha_ml/cli/provision.py:86]: Docstring for raising this exception

File: src/matcha_ml/cli/provision.py [hunk 2]
```diff
@@ -75,6 +75,20 @@ def provision_resources(
...
+        remote_state_manager.remove_matcha_config()
+        print("Synced local state config.")
```
Comment (dudeperf3ct) [inline src/matcha_ml/cli/provision.py:88]: Do we need this line?

File: tests/test_cli/test_provision.py [hunk 3]
```diff
@@ -626,3 +630,60 @@ def test_cli_provision_command_with_provisioned_resources(
     )
 
     assert mock_use_lock.call_count == RUN_TWICE
+
+
+def test_remote_state_removed_when_remote_state_is_stale_with_user_confirmation(
+    runner: CliRunner, matcha_testing_directory: str, mock_use_lock: MagicMock
+):
+    """_summary_.
```
Comment (dudeperf3ct) [inline tests/test_cli/test_provision.py:638]: Docstrings

**Review by dudeperf3ct — CHANGES_REQUESTED** [review]
Implementation looks solid. Few changes regarding missing docstrings.

File: src/matcha_ml/cli/provision.py [hunk 4]
```diff
@@ -75,6 +75,20 @@ def provision_resources(
     """
     remote_state_manager = RemoteStateManager()
 
+    if remote_state_manager.is_state_stale():
+        print(
```
Comment (swells2020) [inline src/matcha_ml/cli/provision.py:79]: Nit: This feels like an important warning message to the user. Could we wrap this in `build_warning_status` from `ui.status_message_builders`?

File: tests/test_cli/test_provision.py [hunk 5]
```diff
@@ -626,3 +630,60 @@ def test_cli_provision_command_with_provisioned_resources(
     )
 
     assert mock_use_lock.call_count == RUN_TWICE
+
+
+def test_remote_state_removed_when_remote_state_is_stale_with_user_confirmation(
+    runner: CliRunner, matcha_testing_directory: str, mock_use_lock: MagicMock
```
Comment (swells2020) [inline tests/test_cli/test_provision.py:636]: Do we need to pass `mock_use_lock` to the test?

**Review by swells2020 — COMMENTED** [review]
Looks great 👍  One nit and one superfluous (I think) argument.

**Review by dudeperf3ct — APPROVED** [review]
LGTM! Awesome @Christopher-Norman

File: src/matcha_ml/cli/provision.py [hunk 6]
```diff
@@ -72,9 +73,25 @@ def provision_resources(
...
+    if remote_state_manager.is_state_stale():
+        print_status(
+            build_warning_status(
+                "Matcha has detected a stale state file. This means that your local configuration is out of sync with the remote state, the resource group may have been removed."
+            )
+        )
+        remove_local_config = typer.confirm(
+            "Do you want to remove the existing local config and continue?"
+        )
+        if not remove_local_config:
```
Comment (JonoCX) [inline src/matcha_ml/cli/provision.py:89]: Nit: could be a one-liner if:

```python
if not typer.confirm("..."):
```

**Review by JonoCX — APPROVED** [review]
Nice work! 🚀  

See couple of nits

**Review by JonoCX — APPROVED** [review]
LGTM

**Pull request merged**